c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1763,6 +2135,146 @@ class DeviceCachingAllocator {
     return blocks;
   }
 
+  // returns the smallest possible address in any segment
+  // where there is enough free address space to fit size
+  // may be composed of free and unmapped segments
+  Block* find_expandable_block(
+      int device,
+      cudaStream_t stream,
+      BlockPool* pool,
+      size_t size) {
+    Block key(device, stream, 0);
+
+    auto allocateable = [](Block* b) {
+      return b && !b->allocated && b->event_count == 0 &&
+          b->stream_uses.empty();
```

Comment:
Why are we checking that `b->stream_uses.empty()` instead of that `b` is on the same stream passed in, as we do for `search_key` ?

Reply:
Everything in a segment is in the same stream. `stream_uses` are other streams that have used this memory that we need to wait for before it becomes ready to be reallocated.